test/lit/passes/cast-refining.wast

```diff
@@ -757,6 +790,29 @@
       )
     )
   )
+
+  ;; YESTNH:      (func $locals (type $none_=>_none)
+  ;; YESTNH-NEXT:  (local $A (ref none))
+  ;; YESTNH-NEXT:  (local $B (ref none))
+  ;; YESTNH-NEXT:  (local $C1 (ref none))
+  ;; YESTNH-NEXT:  (local $C2 nullref)
+  ;; YESTNH-NEXT:  (nop)
+  ;; YESTNH-NEXT: )
+  ;; NO_TNH:      (func $locals (type $none_=>_none)
+  ;; NO_TNH-NEXT:  (local $A (ref none))
+  ;; NO_TNH-NEXT:  (local $B (ref none))
+  ;; NO_TNH-NEXT:  (local $C1 (ref none))
```

Comment:
Is there a pass responsible for optimizing out anything that claims to have an impossible type?

Reply:
In practice GUFA will do so, since no content will arrive there. We could add a small peephole optimization as well for compiler efficiency I suppose.